packages/suite/src/utils/suite/device.ts

```diff
@@ -355,9 +355,15 @@ export const getPhysicalDeviceCount = (devices: Device[]) => {
     return uniqueIds.size;
 };
 
-export const parseFirmwareChangelog = (firmwareRelease: TrezorDevice['firmwareRelease']) => {
+export const parseFirmwareChangelog = (
+    features: TrezorDevice['features'],
+    firmwareRelease: TrezorDevice['firmwareRelease'],
+) => {
+    const supportIntermediary = features?.major_version === 1;
     if (!firmwareRelease?.changelog || firmwareRelease?.changelog?.length === 0) return null;
-    const changelogs = firmwareRelease.changelog;
+    // Generate latest FW changelog if device has intermediary support
+    const latestChangelog = [firmwareRelease.latest];
+    const changelogs = supportIntermediary ? latestChangelog : firmwareRelease.changelog;
 
     // Default changelog format is just a long string where individual changes are separated by "*" symbol.
 
```

Comment:
It could be easily simplified by returning only object of changelog for desired version instead of an array with changelogs for all versions laying between because then in FirmwareOffer `parsedChangelog.slice(0, 1)` is called and only the first item is actually used 🤦 

```suggestion
    const release = firmwareRelease[supportIntermediary ? 'latest' : 'release'];

    // Default changelog format is just a long string where individual changes are separated by "*" symbol.

    const changelog = release.changelog
        .trim()
        .split('*')
        .map(l => l.trim())
        .filter(l => l.length);

    const { url, notes } = release;

    // Get firmware version, convert to string and use it as a key in custom object
    const versionString = release.version.join('.'); // e.g. [1,9,8] => "1.9.8"
    return {
        url,
        notes,
        changelog,
        versionString,
    };
```

so no more need for `.map` over all changelogs and then slicing only the first one...

Reply:
changed in e410946b8e1f2bedae5af3f407396cd634661a78
also changed parsing in `FirmwareOptions` component